### PR TITLE
Homepage message

### PR DESCRIPTION
**Pull request by stanparker**
~~Don't merge this yet.~~ Just showing a basic message showing you what you've searched for. 

[Screenshot here.](https://i.imgur.com/LyCnHJ9.png)

The "if 1 == 1" in there is a placeholder for logic that well either give you a "Welcome" message or your search.

Would also like to format the date.

### CODE REVIEW

**Comment by stanparker** [comment]
[Screenshots of latest commits.] (https://imgur.com/a/t86To), showing a "Welcome" message on the home screen on arrival, and otherwise a nice description of the search.

**Comment by stanparker** [comment]
[Added a warning](https://i.imgur.com/ZOXmjjk.png) about records potentially being slow to open when the search returns more than 350 results.

File: src/client/mtna/archive-template.jade [hunk 1]
```diff
@@ -18,6 +18,10 @@ div.relative(role="main"
   md-content.md-default-theme( flex md-scroll-y elem-click="state.collapse()" )
     // main content
 
+    p.lead(ng-if="state._location._queryString == '' && state._location._startDate == '' && state._location._endDate == '' ") Welcome to the archives! <br /> <br /> To get started, click the blue bar to start a search, or click the "+" icon below to add a record. <br / <br /> <br /> <br /> <small>Hint! You can save your searches by bookmarking the page.</small>
```
Comment (DavidSouther) [inline src/client/mtna/archive-template.jade]: You can make this nicer by adding a . and newline at the end of hte attributes list

```
p.lead(   ).
  Plain Text
```

File: src/client/mtna/archive-template.jade [hunk 2]
```diff
@@ -18,6 +18,10 @@ div.relative(role="main"
...
+    p.lead(ng-if="state._location._queryString != '' || state._location._startDate != '' || state._location._endDate != '' ") You've found <strong>{{ state.records.length }}</strong> record(s) searching for <strong>{{ state._location._queryString == "" ? "all records" : state._location._queryString  }}</strong> from <strong>{{ state._location._startDate == "" ? "the beginning of history" : state._location._startDate | date:'MMM d, y' }}</strong> to <strong>{{ state._location._endDate == "" ? "the end of time" : state._location._endDate | date:'MMM d, y'  }}.</strong>
```
Comment (DavidSouther) [inline src/client/mtna/archive-template.jade]:
```
record{{ state.records.length == 1 ? '' : 's'}}
```

Comment (DavidSouther) [inline src/client/mtna/archive-template.jade]: Move all of the string checks to a getter on the controller:

```
get hasQuery(): boolean {
  return this._location._queryString !== '' ...;
}
```

```
p.lead(ng-if="state.hasQuery")
```

File: src/client/mtna/archive-template.jade [hunk 3]
```diff
@@ -18,6 +18,13 @@ div.relative(role="main"
   md-content.md-default-theme( flex md-scroll-y elem-click="state.collapse()" )
     // main content
 
+    p.lead(ng-if="!state._location.hasSearch && state.records.length == 0").
+      Welcome to the archives! <br /> <br /> To get started, click the blue bar to start a search, or click the "+" icon below to add a record. <br / <br /> <br /> <br /> <small>Hint! You can save your searches by bookmarking the page.</small>
```
Comment (DavidSouther) [inline src/client/mtna/archive-template.jade]:
```
 <br / <br /> <br /> <br />
```

is certainly incorrect.

Comment (stanparker) [inline src/client/mtna/archive-template.jade]: Updated formatting with new push.

**Comment by stanparker** [comment]
Should I be using `state._location.hasSearch()` rather than `state._location.hasSearch` in the template?

File: src/client/mtna/archive-template.jade [hunk 4]
```diff
@@ -18,6 +18,17 @@ div.relative(role="main"
   md-content.md-default-theme( flex md-scroll-y elem-click="state.collapse()" )
     // main content
 
+    div.welcome(ng-if="!state._location.hasSearch && state.records.length == 0")
```
Comment (DavidSouther) [inline src/client/mtna/archive-template.jade]: Don't access _location, here or below. Just use `state.hasSearch`

File: src/client/mtna/archive-template.jade [hunk 5]
```diff
@@ -18,6 +18,17 @@ div.relative(role="main"
...
+      p.lead Welcome to the archives!
+      p.lead To get started, click the blue bar to start a search, or click the "+" icon below to add a record.
+      p.lead
+        small Hint! You can save your searches by bookmarking the page.
+
+    div.welcome(ng-if="state.records.length != 0 || state._location.hasSearch")
+      p.lead You've found <strong>{{ state.records.length }}</strong> record{{ state.records.length == 1 ? '' : 's'}} searching for <strong>{{ state._location._queryString == "" ? "all records" : state._location._queryString  }}</strong> from <strong>{{ state._location._startDate == "" ? "the beginning of history" : state._location._startDate | date:'MMM d, y' }}</strong> to <strong>{{ state._location._endDate == "" ? "the end of time" : state._location._endDate | date:'MMM d, y'  }}.</strong>
```
Comment (DavidSouther) [inline src/client/mtna/archive-template.jade]: Break this line up, try to keep each segment below 80 characters. 
```
p.lead
  | You've found
  strong {{ state.records.length }}
  | record{{...}}}
```

File: src/client/mtna/archive-template.jade [hunk 6]
```diff
@@ -18,6 +18,26 @@ div.relative(role="main"
   md-content.md-default-theme( flex md-scroll-y elem-click="state.collapse()" )
     // main content
 
+    div.welcome(ng-if="!state.hasSearch && state.records.length == 0")
```
Comment (DavidSouther) [inline src/client/mtna/archive-template.jade]: Put the conditions in the same order, here and line 27.

Comment (stanparker) [inline src/client/mtna/archive-template.jade]: Fixed in most recent push.

**Comment by stanparker** [comment]
Pushed the code changes, and implemented looks like:

screen shot 2016-08-25 at 8 49 04 pm

screen shot 2016-08-25 at 8 49 54 pm

**Comment by DavidSouther** [comment]
lgtm

**Comment by stanparker** [comment]
Cool, thanks. Merging now.

**Pull request merged**